cachi2/core/models/input.py

```diff
@@ -77,7 +77,42 @@ def _path_is_relative(cls, path: Path) -> Path:
         return check_sane_relpath(path)
 
 
-class _DNFOptions(pydantic.BaseModel, extra="forbid"):
+class _SSLFields(pydantic.BaseModel, extra="forbid"):
+    """SSL options model.
+
+    Provides the 'ssl' key under 'options' for rpm package manager.
+
+    """
+
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+    ca_bundle: Optional[str] = None
+    ssl_verify: Optional[bool] = None
+
+    @pydantic.model_validator(mode="before")
+    def _validate_ssl_options(cls, data: Any, info: pydantic.ValidationInfo) -> Optional[Dict]:
+
+        if ("client_cert" in data.keys() and "client_key" not in data.keys()) or (
+            "client_key" in data.keys() and "client_cert" not in data.keys()
+        ):
+
+            raise SSLContextError(
+                "A required client credential parameter is missing.",
+                solution=(
+                    "When using client certificates,  client_key and client_cert must both be provided."
+                ),
+            )
+
+        return data
+
+
+class _SSLOptions(pydantic.BaseModel, extra="allow"):
+    """SSL options model."""
+
+    ssl: _SSLFields
+
+
```

Comment:
I just posted #651 which should make your life easier when it comes to adding a new options model, you'll only need `_SSLFields` (renamed to `_SSLOptions`), but not both as it's confusing.